src/Dfc.CourseDirectory.Web/Helpers/VenueHelper.cs

```diff
@@ -19,12 +20,10 @@ public static class VenueHelper
                     {
                         if (!venueNames.ContainsKey((Guid)courseRun.VenueId))
                         {
-                            var result = await venueService.GetVenueByIdAsync(new GetVenueByIdCriteria(courseRun.VenueId.ToString()));
-                            if (result.IsSuccess)
-                            {
-                                Guid.TryParse(result.Value.ID, out Guid venueId);
-                                venueNames.Add(venueId, result.Value.VenueName);
-                            }
+                            var venue = await cosmosDbQueryDispatcher.ExecuteQuery(
```

Comment:
This call could potentially be made quite a few times - why not use the `GetVenuesByProvider` query and do the matching in linq, or better yet, create a `GetVenuesByVenueIds` query to grab everything needed in one hit?

Reply:
Do you want to roll that optimisation into this refactor or do a new PR?

Reply:
It was more of a question of _should we do this?_, rather than a statement of "we _should_ do this" - apologies if that wasn't clear. I've looked at the underlying code and it's doing exactly what you've implemented, albeit via a dumb proxy, so I'm on the fence whether it's worth optimising dying code, especially given that you've technically made it no worse than it already is.